[2025-07-07T06:30:25.088+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-07T06:30:25.107+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: lakehouse_pipeline.bronze_to_silver scheduled__2025-07-06T06:00:00+00:00 [queued]>
[2025-07-07T06:30:25.113+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: lakehouse_pipeline.bronze_to_silver scheduled__2025-07-06T06:00:00+00:00 [queued]>
[2025-07-07T06:30:25.113+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-07-07T06:30:25.121+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): bronze_to_silver> on 2025-07-06 06:00:00+00:00
[2025-07-07T06:30:25.127+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'lakehouse_pipeline', 'bronze_to_silver', 'scheduled__2025-07-06T06:00:00+00:00', '--job-id', '6', '--raw', '--subdir', 'DAGS_FOLDER/pipeline_dag.py', '--cfg-path', '/tmp/tmpcip33ilo']
[2025-07-07T06:30:25.128+0000] {standard_task_runner.py:91} INFO - Job 6: Subtask bronze_to_silver
[2025-07-07T06:30:25.130+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=1375) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-07T06:30:25.130+0000] {standard_task_runner.py:63} INFO - Started process 1376 to run task
[2025-07-07T06:30:25.156+0000] {task_command.py:426} INFO - Running <TaskInstance: lakehouse_pipeline.bronze_to_silver scheduled__2025-07-06T06:00:00+00:00 [running]> on host 823b4fee21e3
[2025-07-07T06:30:25.206+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='lakehouse_pipeline' AIRFLOW_CTX_TASK_ID='bronze_to_silver' AIRFLOW_CTX_EXECUTION_DATE='2025-07-06T06:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-07-06T06:00:00+00:00'
[2025-07-07T06:30:25.206+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-07T06:30:41.134+0000] {python.py:237} INFO - Done. Returned value was: None
[2025-07-07T06:30:41.136+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-07T06:30:41.154+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=lakehouse_pipeline, task_id=bronze_to_silver, run_id=scheduled__2025-07-06T06:00:00+00:00, execution_date=20250706T060000, start_date=20250707T063025, end_date=20250707T063041
[2025-07-07T06:30:41.213+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2025-07-07T06:30:41.231+0000] {taskinstance.py:3498} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-07-07T06:30:41.232+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-07-07T07:25:56.365+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-07T07:25:56.387+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: lakehouse_pipeline.bronze_to_silver scheduled__2025-07-06T06:00:00+00:00 [queued]>
[2025-07-07T07:25:56.393+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: lakehouse_pipeline.bronze_to_silver scheduled__2025-07-06T06:00:00+00:00 [queued]>
[2025-07-07T07:25:56.393+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-07-07T07:25:56.401+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): bronze_to_silver> on 2025-07-06 06:00:00+00:00
[2025-07-07T07:25:56.409+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'lakehouse_pipeline', 'bronze_to_silver', 'scheduled__2025-07-06T06:00:00+00:00', '--job-id', '5', '--raw', '--subdir', 'DAGS_FOLDER/pipeline_dag.py', '--cfg-path', '/tmp/tmp24oe8p5c']
[2025-07-07T07:25:56.410+0000] {standard_task_runner.py:91} INFO - Job 5: Subtask bronze_to_silver
[2025-07-07T07:25:56.410+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=506) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-07T07:25:56.410+0000] {standard_task_runner.py:63} INFO - Started process 508 to run task
[2025-07-07T07:25:56.440+0000] {task_command.py:426} INFO - Running <TaskInstance: lakehouse_pipeline.bronze_to_silver scheduled__2025-07-06T06:00:00+00:00 [running]> on host fb8ac4917e48
[2025-07-07T07:25:56.501+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='lakehouse_pipeline' AIRFLOW_CTX_TASK_ID='bronze_to_silver' AIRFLOW_CTX_EXECUTION_DATE='2025-07-06T06:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-07-06T06:00:00+00:00'
[2025-07-07T07:25:56.502+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-07T07:26:32.444+0000] {python.py:237} INFO - Done. Returned value was: None
[2025-07-07T07:26:32.445+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-07T07:26:32.464+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=lakehouse_pipeline, task_id=bronze_to_silver, run_id=scheduled__2025-07-06T06:00:00+00:00, execution_date=20250706T060000, start_date=20250707T072556, end_date=20250707T072632
[2025-07-07T07:26:32.516+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2025-07-07T07:26:32.537+0000] {taskinstance.py:3498} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-07-07T07:26:32.539+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-07-07T07:45:55.138+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-07T07:45:55.158+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: lakehouse_pipeline.bronze_to_silver scheduled__2025-07-06T06:00:00+00:00 [queued]>
[2025-07-07T07:45:55.164+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: lakehouse_pipeline.bronze_to_silver scheduled__2025-07-06T06:00:00+00:00 [queued]>
[2025-07-07T07:45:55.164+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-07-07T07:45:55.172+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): bronze_to_silver> on 2025-07-06 06:00:00+00:00
[2025-07-07T07:45:55.178+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'lakehouse_pipeline', 'bronze_to_silver', 'scheduled__2025-07-06T06:00:00+00:00', '--job-id', '4', '--raw', '--subdir', 'DAGS_FOLDER/pipeline_dag.py', '--cfg-path', '/tmp/tmp2brz9g8h']
[2025-07-07T07:45:55.179+0000] {standard_task_runner.py:91} INFO - Job 4: Subtask bronze_to_silver
[2025-07-07T07:45:55.181+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=470) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-07T07:45:55.181+0000] {standard_task_runner.py:63} INFO - Started process 471 to run task
[2025-07-07T07:45:55.208+0000] {task_command.py:426} INFO - Running <TaskInstance: lakehouse_pipeline.bronze_to_silver scheduled__2025-07-06T06:00:00+00:00 [running]> on host 2a32f8b8ad06
[2025-07-07T07:45:55.264+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='lakehouse_pipeline' AIRFLOW_CTX_TASK_ID='bronze_to_silver' AIRFLOW_CTX_EXECUTION_DATE='2025-07-06T06:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-07-06T06:00:00+00:00'
[2025-07-07T07:45:55.264+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-07T07:46:11.618+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-07T07:46:11.619+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/bronze_to_silver.py", line 17, in run
    .getOrCreate()
     ^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/sql/session.py", line 477, in getOrCreate
    sc = SparkContext.getOrCreate(sparkConf)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 512, in getOrCreate
    SparkContext(conf=conf or SparkConf())
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 198, in __init__
    SparkContext._ensure_initialized(self, gateway=gateway, conf=conf)
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 432, in _ensure_initialized
    SparkContext._gateway = gateway or launch_gateway(conf)
                                       ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/java_gateway.py", line 106, in launch_gateway
    raise RuntimeError("Java gateway process exited before sending its port number")
RuntimeError: Java gateway process exited before sending its port number
[2025-07-07T07:46:11.638+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=lakehouse_pipeline, task_id=bronze_to_silver, run_id=scheduled__2025-07-06T06:00:00+00:00, execution_date=20250706T060000, start_date=20250707T074555, end_date=20250707T074611
[2025-07-07T07:46:11.657+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 4 for task bronze_to_silver (Java gateway process exited before sending its port number; 471)
[2025-07-07T07:46:11.688+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-07-07T07:46:11.723+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-07T07:46:11.726+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-07-07T10:29:15.686+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-07T10:29:15.705+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: lakehouse_pipeline.bronze_to_silver scheduled__2025-07-06T06:00:00+00:00 [queued]>
[2025-07-07T10:29:15.711+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: lakehouse_pipeline.bronze_to_silver scheduled__2025-07-06T06:00:00+00:00 [queued]>
[2025-07-07T10:29:15.711+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-07-07T10:29:15.719+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): bronze_to_silver> on 2025-07-06 06:00:00+00:00
[2025-07-07T10:29:15.726+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'lakehouse_pipeline', 'bronze_to_silver', 'scheduled__2025-07-06T06:00:00+00:00', '--job-id', '5', '--raw', '--subdir', 'DAGS_FOLDER/pipeline_dag.py', '--cfg-path', '/tmp/tmp0_ocptu6']
[2025-07-07T10:29:15.728+0000] {standard_task_runner.py:91} INFO - Job 5: Subtask bronze_to_silver
[2025-07-07T10:29:15.728+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=579) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-07T10:29:15.728+0000] {standard_task_runner.py:63} INFO - Started process 582 to run task
[2025-07-07T10:29:15.757+0000] {task_command.py:426} INFO - Running <TaskInstance: lakehouse_pipeline.bronze_to_silver scheduled__2025-07-06T06:00:00+00:00 [running]> on host 84547bbd5155
[2025-07-07T10:29:15.813+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='lakehouse_pipeline' AIRFLOW_CTX_TASK_ID='bronze_to_silver' AIRFLOW_CTX_EXECUTION_DATE='2025-07-06T06:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-07-06T06:00:00+00:00'
[2025-07-07T10:29:15.813+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-07T10:29:57.805+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-07T10:29:57.807+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/bronze_to_silver.py", line 17, in run
    .getOrCreate()
     ^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/sql/session.py", line 477, in getOrCreate
    sc = SparkContext.getOrCreate(sparkConf)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 512, in getOrCreate
    SparkContext(conf=conf or SparkConf())
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 198, in __init__
    SparkContext._ensure_initialized(self, gateway=gateway, conf=conf)
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 432, in _ensure_initialized
    SparkContext._gateway = gateway or launch_gateway(conf)
                                       ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/java_gateway.py", line 106, in launch_gateway
    raise RuntimeError("Java gateway process exited before sending its port number")
RuntimeError: Java gateway process exited before sending its port number
[2025-07-07T10:29:57.826+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=lakehouse_pipeline, task_id=bronze_to_silver, run_id=scheduled__2025-07-06T06:00:00+00:00, execution_date=20250706T060000, start_date=20250707T102915, end_date=20250707T102957
[2025-07-07T10:29:57.844+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 5 for task bronze_to_silver (Java gateway process exited before sending its port number; 582)
[2025-07-07T10:29:57.867+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-07-07T10:29:57.900+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-07T10:29:57.903+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-07-07T11:12:56.880+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-07T11:12:56.899+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: lakehouse_pipeline.bronze_to_silver scheduled__2025-07-06T06:00:00+00:00 [queued]>
[2025-07-07T11:12:56.904+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: lakehouse_pipeline.bronze_to_silver scheduled__2025-07-06T06:00:00+00:00 [queued]>
[2025-07-07T11:12:56.904+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-07-07T11:12:56.912+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): bronze_to_silver> on 2025-07-06 06:00:00+00:00
[2025-07-07T11:12:56.920+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=490) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-07T11:12:56.921+0000] {standard_task_runner.py:63} INFO - Started process 493 to run task
[2025-07-07T11:12:56.921+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'lakehouse_pipeline', 'bronze_to_silver', 'scheduled__2025-07-06T06:00:00+00:00', '--job-id', '5', '--raw', '--subdir', 'DAGS_FOLDER/pipeline_dag.py', '--cfg-path', '/tmp/tmpfl5t6khm']
[2025-07-07T11:12:56.922+0000] {standard_task_runner.py:91} INFO - Job 5: Subtask bronze_to_silver
[2025-07-07T11:12:56.951+0000] {task_command.py:426} INFO - Running <TaskInstance: lakehouse_pipeline.bronze_to_silver scheduled__2025-07-06T06:00:00+00:00 [running]> on host cc0e4e7d50eb
[2025-07-07T11:12:57.000+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='lakehouse_pipeline' AIRFLOW_CTX_TASK_ID='bronze_to_silver' AIRFLOW_CTX_EXECUTION_DATE='2025-07-06T06:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-07-06T06:00:00+00:00'
[2025-07-07T11:12:57.000+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-07T11:13:53.319+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-07T11:13:53.321+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/bronze_to_silver.py", line 17, in run
    .getOrCreate()
     ^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/sql/session.py", line 477, in getOrCreate
    sc = SparkContext.getOrCreate(sparkConf)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 512, in getOrCreate
    SparkContext(conf=conf or SparkConf())
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 198, in __init__
    SparkContext._ensure_initialized(self, gateway=gateway, conf=conf)
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 432, in _ensure_initialized
    SparkContext._gateway = gateway or launch_gateway(conf)
                                       ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/java_gateway.py", line 106, in launch_gateway
    raise RuntimeError("Java gateway process exited before sending its port number")
RuntimeError: Java gateway process exited before sending its port number
[2025-07-07T11:13:53.339+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=lakehouse_pipeline, task_id=bronze_to_silver, run_id=scheduled__2025-07-06T06:00:00+00:00, execution_date=20250706T060000, start_date=20250707T111256, end_date=20250707T111353
[2025-07-07T11:13:53.355+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 5 for task bronze_to_silver (Java gateway process exited before sending its port number; 493)
[2025-07-07T11:13:53.360+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-07-07T11:13:53.376+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-07T11:13:53.378+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2025-07-07T11:21:57.499+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-07T11:21:57.530+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: lakehouse_pipeline.bronze_to_silver scheduled__2025-07-06T06:00:00+00:00 [queued]>
[2025-07-07T11:21:57.537+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: lakehouse_pipeline.bronze_to_silver scheduled__2025-07-06T06:00:00+00:00 [queued]>
[2025-07-07T11:21:57.538+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-07-07T11:21:57.547+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): bronze_to_silver> on 2025-07-06 06:00:00+00:00
[2025-07-07T11:21:57.557+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=520) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-07T11:21:57.558+0000] {standard_task_runner.py:63} INFO - Started process 522 to run task
[2025-07-07T11:21:57.557+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'lakehouse_pipeline', 'bronze_to_silver', 'scheduled__2025-07-06T06:00:00+00:00', '--job-id', '4', '--raw', '--subdir', 'DAGS_FOLDER/pipeline_dag.py', '--cfg-path', '/tmp/tmp8s63gbex']
[2025-07-07T11:21:57.559+0000] {standard_task_runner.py:91} INFO - Job 4: Subtask bronze_to_silver
[2025-07-07T11:21:57.591+0000] {task_command.py:426} INFO - Running <TaskInstance: lakehouse_pipeline.bronze_to_silver scheduled__2025-07-06T06:00:00+00:00 [running]> on host c4e8a223c4af
[2025-07-07T11:21:57.645+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='lakehouse_pipeline' AIRFLOW_CTX_TASK_ID='bronze_to_silver' AIRFLOW_CTX_EXECUTION_DATE='2025-07-06T06:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-07-06T06:00:00+00:00'
[2025-07-07T11:21:57.645+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-07T11:22:57.572+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-07T11:22:57.572+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/bronze_to_silver.py", line 17, in run
    .getOrCreate()
     ^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/sql/session.py", line 477, in getOrCreate
    sc = SparkContext.getOrCreate(sparkConf)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 512, in getOrCreate
    SparkContext(conf=conf or SparkConf())
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 198, in __init__
    SparkContext._ensure_initialized(self, gateway=gateway, conf=conf)
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 432, in _ensure_initialized
    SparkContext._gateway = gateway or launch_gateway(conf)
                                       ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/java_gateway.py", line 106, in launch_gateway
    raise RuntimeError("Java gateway process exited before sending its port number")
RuntimeError: Java gateway process exited before sending its port number
[2025-07-07T11:22:57.580+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=lakehouse_pipeline, task_id=bronze_to_silver, run_id=scheduled__2025-07-06T06:00:00+00:00, execution_date=20250706T060000, start_date=20250707T112157, end_date=20250707T112257
[2025-07-07T11:22:57.592+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 4 for task bronze_to_silver (Java gateway process exited before sending its port number; 522)
[2025-07-07T11:22:57.599+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-07-07T11:22:57.612+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-07T11:22:57.614+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
